scipy/sparse/__init__.py

```diff
@@ -84,53 +77,77 @@
    random_array - Random values in a given shape array
    block_array - Build a sparse array from sub-blocks
 
-Building sparse matrices:
+.. _stacking-arrays:
+
+Stacking arrays
+---------------
 
 .. autosummary::
    :toctree: generated/
 
-   eye - Sparse MxN matrix whose k-th diagonal is all ones
-   identity - Identity matrix in sparse matrix format
-   diags - Return a sparse matrix from diagonals
-   spdiags - Return a sparse matrix from diagonals
-   bmat - Build a sparse matrix from sparse sub-blocks
-   random - Random values in a given shape matrix
-   rand - Random values in a given shape matrix (old interface)
+   kron - Kronecker product of two sparse arrays
+   kronsum - Kronecker sum of sparse arrays
+   block_diag - Build a block diagonal sparse array
+   tril - Lower triangular portion of a sparse array
+   triu - Upper triangular portion of a sparse array
+   hstack - Stack sparse arrays horizontally (column wise)
+   vstack - Stack sparse arrays vertically (row wise)
 
-Building larger structures from smaller (array or matrix)
+Sparse tools
+------------
 
 .. autosummary::
    :toctree: generated/
 
-   kron - kronecker product of two sparse matrices
-   kronsum - kronecker sum of sparse matrices
-   block_diag - Build a block diagonal sparse matrix
-   tril - Lower triangular portion of a matrix in sparse format
-   triu - Upper triangular portion of a matrix in sparse format
-   hstack - Stack sparse matrices horizontally (column wise)
-   vstack - Stack sparse matrices vertically (row wise)
+   save_npz - Save a sparse array to a file using ``.npz`` format.
+   load_npz - Load a sparse array from a file using ``.npz`` format.
+   find - Return the indices and values of the nonzero elements
 
-Save and load sparse matrices:
+Identifying sparse arrays
+-------------------------
+
+- use ``isinstance(A, sp.sparse.sparray)`` to check whether an array or matrix.
+- use ``A.format == 'csr'`` to check the sparse format
```

Comment:
This was already there, but ideally it shouldn't be - this belongs in a usage guide. And perhaps also in the docstring of `issparse` as a hint. This page is all function/class listings, these two bullet points are outliers (and they're not much more important than many other "how to use" docs). Let's move this out?

Reply:
I removed the references to non-function idioms in `__init__.py` and added them to the `issparse` doc_string.